code/web/Drivers/Koha.php

```
@@ -4699,7 +4699,7 @@ function selfRegister(): array {
 			$postVariables = $this->setPostFieldWithDifferentName($postVariables, 'gender', 'borrower_sex', $library->useAllCapsWhenSubmittingSelfRegistration);
 			$postVariables = $this->setPostFieldWithDifferentName($postVariables, 'initials', 'borrower_initials', $library->useAllCapsWhenSubmittingSelfRegistration);
 			if (!isset($_REQUEST['borrower_branchcode']) || $_REQUEST['borrower_branchcode'] == -1) {
-				$postVariables['library_id'] = $library->code;
+				$postVariables['library_id'] = $library->getMainLocation()->code;
```

@lathomas64 we should check to see if the library ilsCode is empty before going to the main location. The main location could also be null so we should check that case as well. If we don't have a library code or a main location we should return an error to the user. 

It does look like this was originally checking code rather than ilsCode which was incorrect (since library does not have a code).

---

ilsCode was not working for the partner I had been looking at but it looks like that was a user Error and it should have been blank. updating to look at that field first. We would be getting an error from the processSelfRegistration after the request to Koha failed but updating here to fail earlier and save the trip to Koha servers.